### PR TITLE
UCM: Search for symbols in .got, in additon to .got.plt

### PR DESCRIPTION
**Pull request by yosefe**
Fixes issue with UCX python bindings where dlopen() call from python executable was not hooked, so a library loaded by python (runtime.cpython-37m-x86_64-linux-gnu.so) did not have Cuda hooks installed

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9874/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9882/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9924/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9961/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9976/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9992/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9993/ for details (Mellanox internal link).

File: src/ucm/util/reloc.c [hunk 1]
```diff
@@ -178,37 +199,53 @@ ucm_reloc_modify_got(ElfW(Addr) base, const ElfW(Phdr) *phdr, const char *phname
...
-            /* store default entry to prev_value to guarantee valid pointers
-             * throughout life time of the process */
-            if (ctx->def_dlinfo.dli_fbase == entry_dlinfo.dli_fbase) {
-                ctx->patch->prev_value = *entry;
-                ucm_trace("'%s' prev_value is %p from '%s'", ctx->patch->symbol,
-                          *entry, basename(entry_dlinfo.dli_fname));
-            }
+    /* enable writing to the page */
+    page  = (void *)((intptr_t)entry & ~(page_size - 1));
+    ret = mprotect(page, page_size, PROT_READ|PROT_WRITE);
```
Comment (hoopoepg) [inline src/ucm/util/reloc.c]: minor: unaligned ```=```

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/10004/ for details (Mellanox internal link).

**Pull request merged**